### ISSUE TITLE
Using global var as default param initializer and masking that var breaks rest params

### ISSUE THREAD

**Issue by jahqueel** [babel/babel#1128]
Take a look at what the following code generates:

```javascript
const bar = true;
function foo(a = bar, ...b) {
  const bar = false;   // force generation of IIFE
  console.log(`b[1] should be 3, but it's ${b[1]}`);   // attempt to access arguments
}

foo(1, 2, 3);
```

Here's an excerpt of the output:

```javascript
function foo() {
  var a = arguments[0] === undefined ? bar : arguments[0];
  return (function () {
    var bar = false; // force generation of IIFE
    console.log("b[1] should be 3, but it's " + arguments[2]); // attempt to access arguments
  })();
}
```

Attempting to access the arguments array from the IIFE is obviously not going to work.

**Comment by jahqueel** [comment]
If you replace `function foo()` with `function* foo()`, you get an even more convoluted version of #1016

**Comment by sebmck** [comment]
Sorry for the delay. Fixed as of 5.1.0, thanks!
